src/MoBi.Presentation/Tasks/Interaction/InitialConditionsTask.cs

```diff
@@ -201,8 +202,12 @@ private void updateFormulaUsages(TBuildingBlock buildingBlock, IReadOnlyList<Ini
          // The clone manager can return an existing formula from the formula cache if it is present and equivalent to the formula being cloned
          // This is important for renaming step. We can't just rename this formula if one already exists in the cache with the same name because
          // it might actually be the same one.
+         var formulaCount = buildingBlock.FormulaCache.Count;
          var clonedFormula = _cloneManagerForBuildingBlock.Clone(moleculeBuilder.DefaultStartFormula, buildingBlock.FormulaCache);
 
+         if (formulaCount != buildingBlock.FormulaCache.Count)
```

Comment:
This required a comment for me explaining WHY the count can be different. It looks like it can never be the case but in fact it might be because of the side effect of the cloning. I don't like comment but this is where it is required

Reply:
This condition is to ensure we are not executing this AddEvent twice for the same formula
         Since _cloneManagerForBuildingBlock.Clone(moleculeBuilder.DefaultStartFormula, buildingBlock.FormulaCache)
         has the Cache as input parameter, and it is adding the formula if it is not already there
         We need to check if the count has changed to know if the formula was added
         If the count has changed, then we need to publish the event.
         
         Adding all this comment to the code anyways